backend/src/main.js

```diff
@@ -1,114 +1,178 @@
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
+app.use(passport.initialize());
+
+
+app.get('/home', function (req, res, next) {
+	return res.redirect('/');
+});
 
 app.get('/api/entries', function (req, res, next) {
-    return passEntry.find(function(err, passEntryList){
+    return passport.authenticate('jwt', {session: 'false'}, async function (err, user, info){
         if (err){
+            res.statusCode = 401;
+            return res.send({ error: 'Server error: ' + err });
+        }
+        try {
+            const passEntryList = await passEntry.findAll({ where: {userID: user.ID} });
+            res.statusCode = 201;
+            return res.send(passEntryList);
+        } catch (err) {
             res.statusCode = 500;
             log.error('Internal error(%d): %s',res.statusCode,err.message);
             return res.send({ error: 'Server error' });
         }
-        return res.send(passEntryList);
-    });
+    })(req, res, next);
 });
 
-app.get('/api/entries/:id', function (req, res, next) {
-    passEntry.find({'id': res.params[0]}, function(err, passentry){
-        if (err){
+app.get('/api/entries/:id', async function (req, res, next) {
+    try {
+        const passEntryOne = await passEntry.findOne({ where: {ID: res.params[0]} });
+        return res.send(passEntryOne);
+	} catch (err) {
+		res.statusCode = 500;
+		log.error('Internal error(%d): %s',res.statusCode,err.message);
+		return res.send({ error: 'Server error' });		
+	}
+});
+
+app.post('/api/login', function (req, res, next) {
+    passport.authenticate('local', function (err, user){
+        if (user === false) {
+            res.body = "Login failed";
+            return res.status(401).json({ status: 'error', code: 'unauthorized' });
+        } else {
+            const payload = {
+                _id: user.ID,
+                displayName: user.username,
+                email: user.email
+            };
+            const token = jwt.sign(payload, "mysecretkey");
+            return res.json({user: user.username, access_token: token});
+        }
+    })(req, res, next);
+});
+
+/*app.post('/api/token', async function (req, res, next) {
+    if (req.body.user && req.body.password) {
+        var username = req.body.user;
+        var password = req.body.password;
+        var usertest = user.find(function(u) {
+            return u.email === email && u.password === password;
+        });
+        if (usertest) {
+            var payload = {
+                id: usertest._id,
+                displayName: usertest.username,
+                email: usertest.email
+            };
+            var token = jwt.sign(payload, "mysecretkey");
+            res.json({
+                token: token
+            });
+        } else {
+            res.sendStatus(401);
+        }
+    } else {
+        res.sendStatus(401);
+    }
+});*/
+
+app.delete('/api/token',  function (req, res, next) {
```

Comment:
I think a lot about it, and when i look in the OAuth 2.0, i see that login and token are two different APIs for different functionality. But in our situation we have only two actions and they must be according to REST. But i think, that is so... strange may be, that i must login on /api/token.